Almost Done?

mel-d opened the pull request:
The segmenter and database seem to be working. I still want to add a back button to the homepage.

mel-d committed:
Pages created; some code commented out but Welcome is working.

mel-d committed:
Segmenter works. No database yet.

mel-d committed:
Admin link working and page exists. No table or database yet.

mel-d committed:
Database works--haven't tried to clear out or check for bugs yet.

mel-d committed:
minor output format changes

sumeetjain commented on app/views/main/segment_page.erb:
```diff
@@ -0,0 +1,10 @@
+input = params[:string]
+x = StringSegmenter.new
+x.run_program(input)
+%>
```
Too many lines of Ruby in the view file. Controllers are supposed to prepare information for the view. Remember that if you define a variable in the controller and name it with an `@` in front of it, you can refer to it in the view.

sumeetjain commented on lib/string_segmenter.rb:
```diff
@@ -1,6 +1,7 @@
+require_relative 'dictionary'
+
 class StringSegmenter
-  def initialize(str)
```
Can you please explain why these changes to the StringSegmenter were needed?

mel-d committed:
Less Ruby on erb page

mel-d commented:
I fixed the page that had too much Ruby on it. I changed the initialize method to hot having any parameters because, after Googling around for awhile, I read that it's not necessary in some cases. It solved an error I had been getting to change it this way, so I kept it like that.

mel-d committed:
Back button added

sumeetjain commented:
The `initialize` method taking an argument just means that when you create a new object for that class, you have to pass in a value for that parameter.

So you'd do `StringSegmenter.new("thestringyouwanttosegment")`.

Please revert that class back to how it was and see if you can get the code working using the `StringSegmenter.new` method with a supplied argument.

mel-d committed:
Added parameter to initialize

sumeetjain commented:
The other change you made has not been addressed though:

``` ruby
  def run_program(input)
    @str = input
    final_words
    segmented_output
  end
```

This method did not have a parameter before, and it shouldn't now.

(You're extremely close; so if you find yourself changing more than 7 lines of code, ask for help.)

mel-d committed:
run_program parameter deleted

sumeetjain commented:
:+1:
